http-filter-example/http_filter.cc

```diff
@@ -26,9 +29,45 @@ const std::string HttpSampleDecoderFilter::headerValue() const {
   return config_->val();
 }
 
+int HttpSampleDecoderFilter::headerExtra() const {
+  return config_->extra();
+}
+
 FilterHeadersStatus HttpSampleDecoderFilter::decodeHeaders(RequestHeaderMap& headers, bool) {
-  // add a header
-  headers.addCopy(headerKey(), headerValue());
+
+  // parse config: key is header operation, value is a vector of arguments for the operation
+  std::unordered_map<std::string, std::vector<std::string>> header_ops;
+  std::string header_config = headerValue();
+
+  // Find the position of the first space character
+  size_t spacePos = header_config.find(' ');
+  std::string operation = header_config.substr(0, spacePos);
+  std::string args = header_config.substr(spacePos + 1);
+
+  // Create a string stream to iterate over the arguments
+  std::istringstream iss(args);
```

Comment:
Nit: stringstreams in C++ are usually slower than alternatives. It's also copying the entire string, which isn't necessary. 

Not a huge deal since the parsing doesn't need to be in the request path. But this would be faster as e.g. `find`/`substr` and using `string_view`. Could this maybe use https://github.com/envoyproxy/envoy/blob/cacd32bc/source/common/common/utility.cc#L376 ?

It looks like `common/utility` has a lot of string functions that might be useful in the future